controllers/controlplane_controller.go

```diff
@@ -139,6 +138,49 @@ func (r *ControlPlaneReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error
 		r.Log.Info(fmt.Sprintf("PasswordSecret %s exists", instance.Spec.PasswordSecret))
 	}
 
+	ipsetDetails := common.IPSet{
+		Networks:            instance.Spec.Controller.Networks,
+		Role:                controlplane.Role,
+		HostCount:           controlplane.Count,
+		VIP:                 true,
+		AddToPredictableIPs: false,
+	}
+	ipset, op, err := common.OvercloudipsetCreateOrUpdate(r, instance, ipsetDetails)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	if op != controllerutil.OperationResultNone {
+		r.Log.Info(fmt.Sprintf("IPSet for %s successfully reconciled - operation: %s", instance.Name, string(op)))
+	}
+
+	if len(ipset.Status.HostIPs) != controlplane.Count {
```

Comment:
It probably won't be an issue here as the controlplane.Count can only be 1 or 0, but lets keep it the same as in the other controllers.

Reply:
changed